src/iter.rs

```diff
@@ -133,6 +143,11 @@ impl<'a, R:?Sized+'a, U, F> Iterator for FlatMap<'a, R, U, F>
             self.frontiter = Some(IntoIterator::into_iter((self.f)(self.rng)));
         }
     }
+    
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        // See impl for Map above
+        self.len.map_or((usize::MAX, None), |len| (len, Some(len)))
```

Comment:
It looks like len is just the length of the base iterator, not the whole flat map iteration? I think the default size hint is about all that can be done here.

Reply:
libstd flat_map does something more, I guess one can peek at its implementation.